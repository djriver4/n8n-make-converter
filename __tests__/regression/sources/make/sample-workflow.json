{
  "name": "Sample Make Workflow",
  "flow": [
    {
      "id": "1",
      "name": "HTTP",
      "type": "http",
      "parameters": {
        "url": "https://example.com/api",
        "method": "GET",
        "timeout": "5000"
      },
      "metadata": {
        "designer": {
          "x": 100,
          "y": 100
        }
      }
    },
    {
      "id": "2",
      "name": "JSON",
      "type": "json",
      "parameters": {
        "parsedObject": "{{1.data}}"
      },
      "metadata": {
        "designer": {
          "x": 300,
          "y": 100
        }
      }
    },
    {
      "id": "3",
      "name": "Tools",
      "type": "tools",
      "parameters": {
        "code": "// Process the data\nreturn {\n  result: items[0].data.map(function(item) {\n    return item.value * 2;\n  })\n};"
      },
      "metadata": {
        "designer": {
          "x": 500,
          "y": 100
        }
      }
    }
  ]
} 