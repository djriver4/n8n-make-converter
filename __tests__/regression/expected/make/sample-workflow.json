{
  "name": "Converted from n8n",
  "flow": [
    {
      "id": 1,
      "module": "http:ActionSendData",
      "version": 1,
      "parameters": {},
      "mapper": {
        "url": "https://example.com/api",
        "method": "GET",
        "authentication": "none",
        "options": {
          "timeout": 5000
        }
      },
      "metadata": {
        "designer": {
          "x": 100,
          "y": 200
        }
      },
      "name": "HTTP Request"
    },
    {
      "id": 2,
      "module": "unknown:Module",
      "version": 1,
      "parameters": {},
      "mapper": {
        "property": "data",
        "options": {}
      },
      "metadata": {
        "designer": {
          "x": 300,
          "y": 200
        }
      },
      "name": "JSON Parse"
    },
    {
      "id": 3,
      "module": "code:JavaScriptCode",
      "version": 1,
      "parameters": {},
      "mapper": {
        "functionCode": "// Code that contains complex expressions\nreturn {\n  result: $input.first().json.data.map(item => item.value * 2)\n};"
      },
      "metadata": {
        "designer": {
          "x": 500,
          "y": 200
        }
      },
      "name": "Function"
    }
  ],
  "metadata": {
    "instant": false,
    "version": 1,
    "scenario": {
      "roundtrips": 1,
      "maxErrors": 3,
      "autoCommit": true,
      "autoCommitTriggerLast": true,
      "sequential": false,
      "confidential": false,
      "dataloss": false,
      "dlq": false
    },
    "designer": {
      "orphans": []
    },
    "zone": "eu1.make.com"
  }
}